latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/FigureTool.java

```diff
@@ -42,4 +48,36 @@
      * @return true if the figure caption macro is the last content inside the figure macro content or false otherwise
      */
     boolean isFigureCaptionLast(Block figureCaptionBlock);
+
+    /**
+     * @param figureBlock the figure block for which to get the environment name
+     * @return the name of the environment (e.g., "figure", "block", or another value in case of customization by an
+     *     extension)
+     * @since 1.21
+     */
+    default String getFigureEnvironment(Block figureBlock)
```

Comment:
shouldn't we use a `@Unstable` annotation?

Reply:
Added, thanks.